File: backend/dotnet/ticketflow/TicketFlow.TicketService/Domain/Entities/Ticket.cs
using System;

namespace TicketFlow.TicketService.Domain.Entities
{
    internal class Ticket : ITicket
    {
        public Ticket(int id, int movieId, string buyerEmail, int row, int seat, int price)
        {
            Id = id;
            MovieId = movieId;
            BuyerEmail = buyerEmail;
            Row = row;
            Seat = seat;
            Price = price;
        }

        public int Id { get; }

        public int MovieId { get; }

        public string BuyerEmail { get; private set; }

        public int Row { get; }

        public int Seat { get; }

        public int Price { get; }

        public bool IsOrdered => !string.IsNullOrEmpty(BuyerEmail);

        public void Order(string buyerEmail)
        {
            if (string.IsNullOrEmpty(buyerEmail))
            {
                throw new ArgumentNullException();
            }

            BuyerEmail = buyerEmail;
        }
    }
}

File: backend/java/api_gateway/src/main/java/com/ticketflow/api_gateway/models/ticket_service/Ticket.java
package com.ticketflow.api_gateway.models.ticket_service;

import java.util.Optional;

public class Ticket {
    private Integer id;
    private Integer movieId;
    private String buyerEmail;
    private Integer row;
    private Integer seat;
    private Integer price;

    public Ticket(Integer movieId, Optional<String> optionalBuyerEmail, Integer row, Integer seat, Integer price) {
        this.movieId = movieId;
        this.setBuyerEmail(optionalBuyerEmail);
        this.row = row;
        this.seat = seat;
        this.price = price;
    }

    public Integer getId() {
        return id;
    }

    public Integer getMovieId() {
        return movieId;
    }

    public void setMovieId(Integer movieId) {
        this.movieId = movieId;
    }

    public Optional<String> getBuyerEmail() {
        return Optional.ofNullable(buyerEmail);
    }

    public void setBuyerEmail(Optional<String> optionalBuyerEmail) {
        if (optionalBuyerEmail.isPresent()) {
            this.buyerEmail = optionalBuyerEmail.get();
        }
        else {
            this.buyerEmail = null;
        }
    }

    public Integer getRow() {
        return row;
    }

    public void setRow(Integer row) {
        this.row = row;
    }

    public Integer getSeat() {
        return seat;
    }

    public void setSeat(Integer seat) {
        this.seat = seat;
    }

    public Integer getPrice() {
        return price;
    }

    public void setPrice(Integer price) {
        this.price = price;
    }
}